auxiliary/RewardsDistributor/src/RewardsDistributor.sol

```diff
@@ -11,15 +11,18 @@ import {ERC20Helper} from "@synthetixio/core-contracts/contracts/token/ERC20Help
 import {IERC165} from "@synthetixio/core-contracts/contracts/interfaces/IERC165.sol";
 import {IERC20} from "@synthetixio/core-contracts/contracts/interfaces/IERC20.sol";
 
+import {SafeCastU256, SafeCastI256} from "@synthetixio/core-contracts/contracts/utils/SafeCast.sol";
+
 contract RewardsDistributor is IRewardDistributor {
     error NotEnoughRewardsLeft(uint256 amountRequested, uint256 amountLeft);
     error NotEnoughBalance(uint256 amountRequested, uint256 currentBalance);
 
     using ERC20Helper for address;
+    using SafeCastU256 for uint256;
+    using SafeCastI256 for int256;
 
     address public rewardManager;
     uint128 public poolId;
-    address public collateralType;
```

Comment:
We still want to have per collateral distributors available. Please dont remove this functionality

Reply:
before removal, the storage of the value here was only for verification against when calling the `distributeRewards` function. the `distributeRewards` function below still takes in `collateralType` as an argument, allowing for collateral to be distributed to any vault in the pool or at the pool which it is registered to on the core system. since it was only used for that verification before, the storage has been removed because the value isnt used anywhere anymore.

Reply:
Ok makes sense thanks 🙏